be/src/exec/pipeline/nljoin/spillable_nljoin_probe_operator.cpp

```diff
@@ -0,0 +1,255 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "exec/pipeline/nljoin/spillable_nljoin_probe_operator.h"
+
+#include <glog/logging.h>
+
+#include <memory>
+
+#include "common/statusor.h"
+#include "exec/spill/common.h"
+#include "exec/spill/spiller_factory.h"
+
+namespace starrocks::pipeline {
+
+NLJoinProber::NLJoinProber(TJoinOp::type join_op, const std::vector<ExprContext*>& join_conjuncts,
```

Comment:
it seems confusing,  why not name it SpillableNLJoinProber? if both NLJoinProbeOperator and SpillableNLJoinProbeOperator share this common class, put this class in another cpp/h file

Reply:
Our goal is to be able to replace NLJoinProbeOperator, but not all features are currently supported. otherwise it may cause some confusion to the reviewer

Reply:
we could do it when we support all join methods